### DISCUSSION TITLE
How to have table name in result of query

### DISCUSSION THREAD

**Discussion by Gouttfi** [sidorares/node-mysql2#2617] [Q&A] [answered]
Hi, 
I have a little problem, I have several tables where the column names are sometimes identical, I will take a very simple example:
`SELECT t1.id, t2.id FROM t1,t2`
If I run this query, the two ids columns will merge and the result will look something like this :
`[ {id:1} ]`

My question is : Is it possible to have a result that preserves the table names so as not to merge identical column names ?
The solution could be to name the columns with aliases to avoid ambiguity, like `SELECT t1.id AS idT1, t2.id AS idT2 FROM t1,t2`, but in practice I have a lot of columns that have the same name and that forces me to write a query where the SELECT is very long, and makes my code less easy to read and process later.

**Comment by sidorares** [comment, marked as answer]
have you tried `nestTables`? https://github.com/mysqljs/mysql?tab=readme-ov-file#joins-with-overlapping-column-names

@wellwelwel for some reason can't find in the documentation, not sure if it was never there or lost along the way

**Reply by wellwelwel** [reply]
I don't think it ever was, because I also missed `typeCast` in the documentation when I was working on it (where I searched for `nestTables`) 🙋🏻‍♂️

**Reply by Gouttfi** [reply]
@sidorares it seems to be exactly the things I need ! I'm going to do some tests to check this, and will mark this comment as an answer if it ever answers my problem!
Thanks !
